src/Schema/migrations/index.js

```diff
@@ -83,51 +78,56 @@ export type SchemaMigrations = $Exact<{
-    invariant(typeof currentVersion === 'number', 'Current schema version missing in migrations')
-    invariant(minimumVersion > 0, 'Minimum version must be at least 1')
-    invariant(
-      currentVersion >= minimumVersion,
-      'Current schema version must be greater than minimum migrable version',
-    )
-
     invariant(Array.isArray(migrations), 'Missing migrations array')
 
     // validate migrations format
     migrations.forEach(migration => {
       invariant(isObject(migration), `Invalid migration (not an object) in schema migrations`)
-      const { from, to, steps } = migration
+      const { toVersion, steps } = migration
+      invariant(typeof toVersion === 'number', 'Invalid migration - `toVersion` must be a number')
       invariant(
-        typeof from === 'number' && typeof to === 'number' && to > from,
-        'Invalid migration - `to` version must be greater than `from` version',
+        toVersion >= 2,
+        `Invalid migration to version ${toVersion}. Minimum possible migration version is 2`,
       )
       invariant(
         Array.isArray(steps) && steps.every(step => typeof step.type === 'string'),
-        `Invalid migration steps for migration from version ${from} to ${to}. 'changes' should be an array of migration step calls`,
+        `Invalid migration steps for migration to version ${toVersion}. 'steps' should be an array of migration step calls`,
       )
     })
+  }
 
-    // validate if migration spec actually covers every schema version it says it supports
-    const chronologicalMigrations = [...migrations].reverse()
-    let maxCoveredRange = minimumVersion
+  const sortedMigrations = sortMigrations(migrations)
+  const oldestMigration = head(sortedMigrations)
+  const newestMigration = last(sortedMigrations)
+  const minVersion = oldestMigration ? oldestMigration.toVersion - 1 : 1
+  const maxVersion = newestMigration?.toVersion || 1
 
-    chronologicalMigrations.forEach(({ from, to }) => {
-      invariant(
-        maxCoveredRange === from,
-        `Invalid migrations! schemaMigrations() says it covers schema versions from ${minimumVersion} to ${currentVersion}, but there is no listed migration from ${maxCoveredRange} to ${from}. Remember that migrations must be listed in reverse chronological order`,
-      )
-      maxCoveredRange = to
+  if (isDevelopment) {
+    // validate that migration spec is without gaps and duplicates
+    let maxCoveredVersion: ?number = null
+
+    sortedMigrations.forEach(migration => {
+      const { toVersion } = migration
+      if (maxCoveredVersion) {
+        invariant(
+          toVersion === maxCoveredVersion + 1,
+          `Invalid migrations! Migrations listed cover range from version ${minVersion} to ${maxCoveredVersion}, but migration ${JSON.stringify(
+            migration,
+          )} is to version ${toVersion}. Migrations must be listed without gaps, or duplicates.`,
+        )
+      }
+      maxCoveredVersion = toVersion
```

Comment:
I guess we don't need the `maxCoveredVersion` variable, use the `reduce` function instead of `forEach`, please.

Reply:
Oh, Marcin, you… never miss an opportunity to inject a `reduce` 😂